### ISSUE TITLE
Can't setup a functional queue with multiple processes

### ISSUE THREAD

**Issue by vmgustavo** [peter-wangxu/persist-queue#136]
I am trying to use this library in the following setting:
* Script to GET from the Queue
* Script to PUT into Queue

This PUTs
```python
from time import sleep
from persistqueue import Queue

q = Queue('queue', autosave=False, tempdir='tempq')

for idx in range(5):
    sleep(4)
    q.put(idx)
```

This GETs
```python
from time import sleep
from persistqueue import Queue

q = Queue('queue', autosave=False, tempdir='tempq')

while True:
    if not q.empty():
        print('get new')
        obj = q.get()
        sleep(5)
        q.task_done()
    else:
        print('empty q | sleep 2s')
        sleep(2)
```

If I start the PUT script first and later the GET script everything works fine if a put is called before a get. However after the end of the for loop in the PUT script if I restart the script the GET script gets stuck into the else, meaning the queue it is reading is empty. Is this expected?

Another problem (but maybe the same reason that causes the first) is if I start the GET script before the PUT script then the GET script gets stuck into the else the same way as before.

I've tried many different arguments to see if there is any change but no matter what I tried the result is the same.

Is it possible to make this work? If yes what am I doing wrong?

**Comment by bigtonylewis** [comment]
I think I have the same issue.  I'm looking for a persistent, file-based queue where one or more processes can write items to the queue, and one or more processes can read.  If I have a single `put()`ing process and a single `get()`ing process, it works until the `get()`ing process loops back for another time to see what's new in the queue.

Does or should `persist-queue` handle this OK?

**Comment by bigtonylewis** [comment]
Just saw #132, indicating files and sqlite can't be used across processes

**Comment by peter-wangxu** [comment]
A new mysql based queue has been implemented in version 0.8.0, please give a try.

**Comment by steve8210** [comment]
> A new mysql based queue has been implemented in version 0.8.0, please give a try.

Hi, which of queue type form persistqueue is backed by this?
